### PR TITLE
librbd: race condition possible when validating RBD pool

### PR DESCRIPTION
**Pull request by dillaman**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by dillaman** [comment]
retest this please

File: src/librbd/image/CreateRequest.cc [hunk 1]
```diff
@@ -276,80 +277,19 @@ void CreateRequest<I>::validate_data_pool() {
...
-
-  m_outbl.clear();
-  int r = m_data_io_ctx.aio_operate(RBD_INFO, comp, &op, &m_outbl);
-  ceph_assert(r == 0);
-  comp->release();
+  auto ctx = create_context_callback<
+    CreateRequest<I>, &CreateRequest<I>::handle_validate_data_pool>(this);
+  auto req = ValidatePoolRequest<I>::create(m_data_io_ctx, m_op_work_queue,
+                                            ctx);
+  req->send();
```
Comment (trociny) [inline src/librbd/image/CreateRequest.cc]: Just thinking. It looks like we are doing this validation for every namespace. Don't we want to improve it and validate it for no namespace `m_data_io_ctx`? Or probably `ValidatePoolRequest` could dup the `io_cxt` and reset its namespace.

Comment (dillaman) [reply src/librbd/image/CreateRequest.cc]: :+1:

**Comment by dillaman** [comment]
updated

**Pull request merged**